src/celestia/hud.cpp

```diff
@@ -0,0 +1,1226 @@
+        parent = parent.parent();
+
+    if (parent.star() != nullptr)
+    {
+        bool showPhaseAngle = false;
+
+        Star* sun = parent.star();
+        if (sun->getVisibility())
+        {
+            showPhaseAngle = true;
+        }
+        else if (const auto* orbitingStars = sun->getOrbitingStars(); orbitingStars != nullptr && orbitingStars->size() == 1)
+        {
+            // The planet's orbit is defined with respect to a barycenter. If there's
+            // a single star orbiting the barycenter, we'll compute the phase angle
+            // for the planet with respect to that star. If there are no stars, the
+            // planet is an orphan, drifting through space with no star. We also skip
+            // displaying the phase angle when there are multiple stars (for now.)
+            sun = orbitingStars->front();
+            showPhaseAngle = sun->getVisibility();
+        }
+
+        if (showPhaseAngle)
+        {
+            Eigen::Vector3d sunVec = Selection(&body).getPosition(t).offsetFromKm(Selection(sun).getPosition(t));
+            sunVec.normalize();
+            double cosPhaseAngle = std::clamp(sunVec.dot(viewVec.normalized()), -1.0, 1.0);
+            double phaseAngle = acos(cosPhaseAngle);
+            overlay.printf(_("Phase angle: %.1f%s\n"), celmath::radToDeg(phaseAngle), UTF8_DEGREE_SIGN);
+        }
+    }
+
+    if (detail > 1)
+    {
+        if (body.getRotationModel(t)->isPeriodic())
+            displayRotationPeriod(overlay, body.getRotationModel(t)->getPeriod());
+
+        if (body.getName() != "Earth" && body.getMass() > 0)
+            displayMass(overlay, body.getMass(), hudSettings.measurementSystem);
+
+        if (float density = body.getDensity(); density > 0)
+        {
+            if (hudSettings.measurementSystem == MeasurementSystem::Imperial)
+                overlay.printf(_("Density: %.2f x 1000 lb/ft^3\n"), density / (float) OneLbPerFt3InKgPerM3 / 1000.0f);
+            else
+                overlay.printf(_("Density: %.2f x 1000 kg/m^3\n"), density / 1000.0f);
+        }
+
+        float planetTemp = body.getTemperature(t);
+        if (planetTemp > 0)
+            overlay.printf(_("Temperature: %s\n"), KelvinToStr(planetTemp, 3, hudSettings.temperatureScale));
+    }
+}
+
+void
+displayLocationInfo(Overlay& overlay,
+                    const Location& location,
+                    double distanceKm,
+                    MeasurementSystem measurement)
+{
+    overlay.printf(_("Distance: %s\n"), DistanceKmToStr(distanceKm, 5, measurement));
+
+    const Body* body = location.getParentBody();
+    if (body == nullptr)
+        return;
+
+    Eigen::Vector3f locPos = location.getPosition();
+    Eigen::Vector3d lonLatAlt = body->cartesianToPlanetocentric(locPos.cast<double>());
+    displayPlanetocentricCoords(overlay, *body,
+                                lonLatAlt.x(), lonLatAlt.y(), lonLatAlt.z(), false, measurement);
+}
+
+std::string
```

Comment:
```suggestion
std::string /* TODO: replace with std::string_view */
```

Reply:
Simple `string_view` doesn't work here because the star branch can generate a new string if the star has only catalog numbers. Would probably have to be something along the lines of a `std::variant<std::string, std::string_view>`